app/fee_distribution.go

```diff
@@ -69,17 +69,25 @@ func (vac *ValAddrCache) GetDistributionAddr(ctx sdk.Context, consAddr sdk.ConsA
 }
 
 func distributeFeeBEP159(ctx sdk.Context, am auth.AccountKeeper, valAddrCache *ValAddrCache, publishBlockFee bool, stakeKeeper stake.Keeper) (blockFee pub.BlockFee) {
-	fee := fees.Pool.BlockFees()
-	ctx.Logger().Info("FeeCalculation distributeFeeBEP159", "height", ctx.BlockHeader().Height, "fee", fee)
 	blockFee = pub.BlockFee{Height: ctx.BlockHeader().Height}
-	if fee.IsEmpty() {
-		// no fees in this block
+	prevBlockFee := stakeKeeper.BankKeeper.GetCoins(ctx, stake.FeeCollectorAddr)
+	prevProposerDistributionAddr := stakeKeeper.GetPrevProposerDistributionAddr(ctx)
+	ctx.Logger().Info("FeeCalculation distributeFeeBEP159", "height", ctx.BlockHeader().Height, "prevBlockFee", prevBlockFee)
+	// set fee in current block
+	currentBlockFee := fees.Pool.BlockFees().Tokens
+	if !currentBlockFee.IsZero() {
+		if _, _, err := stakeKeeper.BankKeeper.AddCoins(ctx, stake.FeeCollectorAddr, currentBlockFee); err != nil {
+			panic(err)
+		}
+		proposerValAddr := ctx.BlockHeader().ProposerAddress
+		proposerDistributionAddr := valAddrCache.GetDistributionAddr(ctx, proposerValAddr)
```

Comment:
It is possible `GetDistributionAddr` may panic, if the validator changes the `Consensus` address or become no validator.